2003-04-17  Richard Henderson  <rth@redhat.com>

	* config/ia64/ia64.c (ia64_in_small_data_p): Disallow strings.

--- gcc-3.3.1/gcc/config/ia64/ia64.c.ia64-symbol_ref_flags	2003-07-16 12:05:41.000000000 +0200
+++ gcc-3.3.1/gcc/config/ia64/ia64.c	2003-07-16 13:58:15.000000000 +0200
@@ -7263,6 +7263,10 @@ ia64_in_small_data_p (exp)
   if (TARGET_NO_SDATA)
     return false;
 
+  /* We want to merge strings, so we never consider them small data.  */
+  if (TREE_CODE (exp) == STRING_CST)
+    return false;
+
   if (TREE_CODE (exp) == VAR_DECL && DECL_SECTION_NAME (exp))
     {
       const char *section = TREE_STRING_POINTER (DECL_SECTION_NAME (exp));
